### DISCUSSION TITLE
How to jump to the next file header when viewing a multi-file diff through less?

### DISCUSSION THREAD

**Discussion by voidus** [dandavison/delta#1868] [Q&A] [answered]
I'm not sure if this is possible, but it would be nice to jump between file headers in less.

Maybe there is some search magic that would work here? I used `/^diff` before I used delta, but delta doesn't put that marker in.

**Comment by dandavison** [comment, marked as answer]
Hi @voidus, you've seen the [navigate](https://dandavison.github.io/delta/navigation-keybindings-for-large-diffs.html) feature, right?

```
      --navigate
          Activate diff navigation.

          Use n to jump forwards and N to jump backwards. To change the file labels used see
          --file-added-label, --file-copied-label, --file-modified-label, --file-removed-label,
          --file-renamed-label.

      --navigate-regex <REGEX>
          Regular expression defining navigation stop points
```

So what you're aiming for is to use those settings to cause it to visit precisely the boundaries you want. Let me know what you come up with. I think that that particular requirement (file boundaries, but not within-file commit hunk boundaries) is one that's less obvious how to achieve. I believe that for this reason I personally sometimes set the diff context to a high number (e.g. `-U999`) so that there are no within-file hunks.

**Comment by voidus** [comment]
No I didn't, that's why I wrote this. Thank you!
